lib/ex_aws/operation/s3.ex

```diff
@@ -50,11 +49,6 @@ defmodule ExAws.Operation.S3 do
       fun.(config)
     end
 
-    def add_bucket_to_path(operation) do
```

Comment:
I think we should keep this function and make it optional by configuration, so this is not a breaking change.

```elixir
    def maybe_add_bucket_to_path(operation, true = _virtual_host), do: operation

    def maybe_add_bucket_to_path(operation, false = _virtual_host) do
      path = "/#{operation.bucket}/#{operation.path}" |> String.trim_leading("//")
      operation |> Map.put(:path, path)
    end
```

fetching the `virtual_host` from configuration and setting it by default to `true`

Reply:
There is no reason to keep the non-virtual-host style. Instead, `add_bucket_to_path/1` should be kept *as is* except instead of adding the bucket to the *path* it should be adding the bucket to the *host*. 

That would still be a breaking change, however, as existing configurations that currently work would no longer work (including my own apps using this library, so ... :/).